src/CMakeLists.txt

```diff
@@ -146,23 +146,23 @@ install(CODE "execute_process(COMMAND ${CMAKE_COMMAND}
 if (WIN32)
     if (NOT MINGW)
         set(prefix "")
-        set(ext "lib")
+        set(ext ".lib")
```

Comment:
How about using `TARGET_LINKER_FILE_SUFFIX` on CMake 3.15+, and fallback to use `SUFFIX` property on older version?

https://cmake.org/cmake/help/v3.15/manual/cmake-generator-expressions.7.html
https://cmake.org/cmake/help/v3.0/prop_tgt/SUFFIX.html#prop_tgt:SUFFIX

Reply:
It'll really fallback? I did it first, error reported below cmake 3.15.

Reply:
Thanks for the comment @xkszltl! The library supports pretty old Cmake 2.8.11+ (see the [requirements](https://github.com/oneapi-src/oneDNN#requirements-for-building-from-source)), so if we go this route we will need to use the latter approach. I don't mind it, but won't insist on changing the PR either as this is really a legacy piece of code. Intel MKL-DNN was renamed to DNNL 1 year ago (and no new functionality was added to the `mkldnn::` API since then), so I doubt it is used by anyone now. We will probably soon drop it altogether (in [v2.0](https://github.com/oneapi-src/oneDNN/milestone/7)?).

However, if you see the cases where using property would help us avoiding any further issues (like the one you faced in PyTorch), please let us know. That would be good enough justification for switching to more cmake-like way of dealing with the extension.

Reply:
Oh, I see, could try import_suffix.

Reply:
It's more of a FYI when I saw this messy if clause.
Usually when I face similar case I simply go with the modern way (to offloading cross-platform work to cmake itself), and throw those messy part into a version-gated if.

SUFFIX and IMPORT_SUFFIX are also available in 2.8:
https://cmake.org/cmake/help/v2.8.12/cmake.html#prop_tgt:SUFFIX

Reply:
Also I don't see any reason of deprecating compat_lib, if it can be done right within a few lines of code (that's also the reason why I mentioned TARGET_LINKER_FILE_SUFFIX/SUFFIX).
Unless you have some telemetry, it's hard to tell whether there're `-lmkldnn` in other repo (yep, we have -_-|||)

Reply:
> yep, we have -_-

So, you use the `dnnl` header files, but link against `mkldnn`? Or you completely rely on `mkldnn` including outdated headers?

Reply:
No we don't use mkldnn directly but there're various places where mkldnn has to be in linking command or packaging/deployment scripts.
Hard to find them all at once and clean up.

IMHO rebranding is "OK" and throw away backward compatibility is "OK" when it is too hard to maintain, but intentionally breaking backward compatibility is bad. That's why I'm proposing ways to make aliasing code as simple as possible so that it can stay there forever without increasing maintenance cost.

Reply:
I see. Thank you for the feedback!
Given that v2.0 doesn't break the API with older Intel MKL-DNN version (except for the renaming that happened long time ago), we might keep it for a while. But in the long term we would love people to migrate to DNNL library completely, because the day we break the API / ABI in non-compatible way, the `mkldnn` compatibility layer will go away. This might cause problems for those who still use it.

Reply:
JFYI, I tried using:
``` cmake
get_target_property(ext ${LIB_NAME} IMPORT_SUFFIX)
get_target_property(ext ${LIB_NAME} SUFFIX)
# and few others
```
instead of current hard-coded values for extensions, but none of them worked (tested with CMake 3.1.0 and MSVC 2015). Strangely enough, I had bad luck querying many properties listed in cmake documentation. Seems, if used by default, the properties remain unset (?).
Anyways, this seems aligned with https://github.com/oneapi-src/oneDNN/pull/806#discussion_r469438231. So, I am going to promote the PR as is.

Reply:
Maybe SUFFIX/IMPORT_SUFFIX only work for overwriting cmake default in set_target_property

Reply:
Yeah, that's the impression I got too...